react-catalog/my-app/src/Details.tsx

```diff
@@ -13,42 +14,50 @@ export function Details() {
         const currentProduct = await readProduct(Number(productId));
         setProduct(currentProduct);
       } catch (error) {
-        console.error(error);
+        alert(error);
       }
     }
     productCall();
   }, [productId]);
+
+  function handleAddToCart() {
+    alert(`Added ${product?.name} to cart`);
+    navigate('/');
+  }
+
   return (
     <>
-      <div className="flex flex-col border-2 rounded-b-lg p-4">
-        <Link to="/" className="self-start">
-          {' '}
-          {'< Back to catalog'}{' '}
-        </Link>
-        <div className="flex">
-          <div className="flex justify-center items-center w-[316px] h-[316px]">
-            <img
-              src={product?.imageUrl}
-              className="max-w-[316px] max-h-[316px]"></img>
-          </div>
-          <div className="m-4">
-            <h3 className="text-[22px] font-semibold">{product?.name}</h3>
-            <h6 className="text-gray-400 text-[16px] font-semibold">{`$${String(
-              product?.price
-            ).slice(0, -2)}.${String(product?.price).slice(-2)}`}</h6>
-            <p className="text-[12px]">{product?.shortDescription}</p>
+      {product ? (
+        <>
+          <div className="flex flex-col border-2 rounded-b-lg p-4">
+            <Link to="/" className="self-start">
+              &lt; Back to catalog
+            </Link>
+            <div className="flex">
+              <div className="flex justify-center items-center w-[316px] h-[316px]">
+                <img
+                  src={product.imageUrl}
+                  className="max-w-[316px] max-h-[316px]"></img>
+              </div>
+              <div className="m-4">
+                <h3 className="text-[22px] font-semibold">{product.name}</h3>
+                <h6 className="text-gray-400 text-[16px] font-semibold">
+                  {toDollars(Number(product.price))}
+                </h6>
+                <p className="text-[12px]">{product.shortDescription}</p>
+              </div>
+            </div>
+            <p>{product.longDescription}</p>
           </div>
-        </div>
-        <p>{product?.longDescription}</p>
-      </div>
-      <button
-        className="p-2 bg-[#F7F5F8] hover:outline hover:bg-[#E9E8EA] outline-2 rounded-lg"
-        onClick={() => {
-          alert(`Added ${product?.name} to cart`);
-          navigate('/');
-        }}>
-        Add to Cart
-      </button>
+          <button
+            className="p-2 bg-[#F7F5F8] hover:outline hover:bg-[#E9E8EA] outline-2 rounded-lg"
+            onClick={handleAddToCart}>
+            Add to Cart
+          </button>
+        </>
+      ) : (
+        <p className="p-4">Loading...</p>
```

Comment:
Oh, this is how you are doing the Loading message. This mostly works. However, if there is an error, `product` will still be undefined so you will get the Loading message, even though the loading finished with an error. That might still confuse the user.